core/src/main/java/org/mskcc/cbio/portal/scripts/UpdateGenePanel.java

```diff
@@ -0,0 +1,145 @@
+
+import joptsimple.*;
+
+import static org.mskcc.cbio.portal.util.GenePanelUtil.extractGenes;
+import static org.mskcc.cbio.portal.util.GenePanelUtil.extractPropertyValue;
+
+/**
+ * Updates an existing, potentially in use gene panel.
+ *
+ * @author jtquach1
+ */
+public class UpdateGenePanel extends ConsoleRunnable {
+
+    private File genePanelFile;
+    private static final Scanner s = new Scanner(System.in);
+
+    @Override
+    public void run() {
+        try {
+            String progName = "UpdateGenePanel";
+            String description = "Update gene panel files.";
+
+            OptionParser parser = new OptionParser();
+            OptionSpec<String> data = parser
+                .accepts("data", "gene panel file")
+                .withRequiredArg()
+                .describedAs("data_file.txt")
+                .ofType(String.class);
+            parser.accepts("noprogress",
+                "this option can be given to avoid the messages regarding memory usage and % complete");
+
+            OptionSet options;
+            try {
+                options = parser.parse(args);
+            } catch (OptionException e) {
+                throw new UsageException(progName, description, parser, e.getMessage());
+            }
+            File genePanel_f;
+            if (options.has(data)) {
+                genePanel_f = new File(options.valueOf(data));
+            } else {
+                throw new UsageException(progName, description, parser, "'data' argument required.");
+            }
+
+            setFile(genePanel_f);
+            importData();
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void importData() throws Exception {
+        ProgressMonitor.setCurrentMessage("Reading data from:  " + genePanelFile.getAbsolutePath());
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(genePanelFile));
+
+        ProgressMonitor.setCurrentMessage("Retrieving matching gene panel from database.");
+        String stableId = extractPropertyValue("stable_id", properties, true);
+        Set<CanonicalGene> canonicalGenes = extractGenes(properties, true);
+        GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(stableId);
+
+        if (genePanel == null) {
+            ProgressMonitor.logWarning("Gene panel " + stableId + " does not exist in the database! Exiting.");
+            return;
+        }
+
+        System.err.println("WARNING: You are about to make the following changes to gene panel " + stableId + ":");
+        DaoGenePanel.updatePreview(genePanel, canonicalGenes);
+        System.err.println("Proceed? This action is irreversible. (Y/n)");
+
+        if (promptUser()) {
+            if (DaoSampleProfile.sampleProfileMappingExistsByPanel(genePanel.getInternalId())) {
+                ProgressMonitor.logWarning("Gene panel " + stableId
```

Comment:
Shouldn't this be logDebug instead of logWarning?
```suggestion
                ProgressMonitor.logDebug("Gene panel " + stableId
```

Reply:
hm, I suppose so since it's not a "warning" the user has to be worried about on their end, but I'm not sure whether to use setCurrentMessage over logDebug